### PR TITLE
[SPARK-42756][CONNECT][PYTHON] Helper function to convert proto literal to value in Python Client

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Helper function to convert proto literal to value in Python Client

### Why are the changes needed?
needed in .ml

### Does this PR introduce _any_ user-facing change?
no, dev-only

### How was this patch tested?
added ut

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @WeichenXu123 @HyukjinKwon

File: python/pyspark/sql/connect/expressions.py [hunk 1]
```diff
@@ -308,6 +308,43 @@ def _infer_type(cls, value: Any) -> DataType:
     def _from_value(cls, value: Any) -> "LiteralExpression":
         return LiteralExpression(value=value, dataType=LiteralExpression._infer_type(value))
 
+    @classmethod
+    def _to_value(cls, literal: "proto.Expression.Literal") -> Any:
+        if literal.HasField("null"):
+            return None
```
Comment (WeichenXu123) [inline python/pyspark/sql/connect/expressions.py:314]: Shall we raise error in this case ?

File: python/pyspark/sql/connect/expressions.py [hunk 2]
```diff
@@ -308,6 +308,43 @@ def _infer_type(cls, value: Any) -> DataType:
...
+        elif literal.HasField("date"):
+            return DateType().fromInternal(literal.date)
+        elif literal.HasField("timestamp"):
+            return TimestampType().fromInternal(literal.timestamp)
+        elif literal.HasField("timestamp_ntz"):
+            return TimestampNTZType().fromInternal(literal.timestamp_ntz)
+        elif literal.HasField("day_time_interval"):
+            return DayTimeIntervalType().fromInternal(literal.day_time_interval)
+        elif literal.HasField("array"):
+            return [LiteralExpression._to_value(v) for v in literal.array.elements]
```
Comment (WeichenXu123) [inline python/pyspark/sql/connect/expressions.py:344]: Q: shall we verify the element_type field ?

Comment (zhengruifeng) [reply python/pyspark/sql/connect/expressions.py:344]: that is fine

File: python/pyspark/sql/connect/expressions.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/expressions.py:314]: `None` here is a valid value.

File: python/pyspark/sql/connect/expressions.py (hunk 2 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/expressions.py:344]: let me add one

**Comment by xinrong-meng** [comment]
Do we intentionally ignore `YearMonthIntervalType`?

**Review by xinrong-meng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
> Do we intentionally ignore `YearMonthIntervalType`?

it seems that we have not support `YearMonthIntervalType` in vanilla PySpark, and the Python Client is reusing PySpark's types.
So it seems that we should add `YearMonthIntervalType` in PySpark first

**Comment by zhengruifeng** [comment]
thank you all, merged to master/branch-3.4

**Pull request closed without merging**